diff -uNr a/src/xorg.conf b/src/xorg.conf
--- a/src/xorg.conf	2015-01-13 19:23:42.000000000 -0800
+++ b/src/xorg.conf	2015-12-03 02:28:52.422148387 -0800
@@ -1,17 +1,21 @@
 # X.Org X server configuration file for xfree86-video-mali
 
+Section "Module"
+        Load            "exa"
+        Load            "fb"
+EndSection
+
 Section "Device"
         Identifier      "Mali-Fbdev"
         Driver          "mali"
-        Option          "fbdev"         "/dev/fb0"
+        Option          "fbdev"           "/dev/fb0"
         Option          "DRI2"            "true"
         Option          "DRI2_PAGE_FLIP"  "false"
         Option          "DRI2_WAIT_VSYNC" "false"
         Option          "UMP_CACHED"      "true"
-        Option          "UMP_LOCK"        "false"
+        Option          "UMP_LOCK"        "true"
         Option          "SWCursor"        "true"
         Option          "HWCursor"        "false"
-
 EndSection
 
 Section "ServerFlags"
